src/cli/generator/config.rs

```diff
@@ -23,6 +23,8 @@ pub struct Config<Status = UnResolved> {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub preset: Option<PresetConfig>,
     pub schema: Schema,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub secret: Option<String>,
```

Comment:
Need to abstract this out using a NewType

Reply:
Instead of using `String` use 

```rust
struct TemplateString(Mustache);

```
Implement it's serde so that you can simply plug and play TemplateString at the time of deserialization instead of introducing an additional step.